src/main/java/com/google/javascript/clutz/DeclarationGenerator.java

```diff
@@ -1418,28 +1435,40 @@ private boolean isNewableFunctionType(FunctionType type) {
      * TypeScript does not have an easy one-liner syntax. Instead we have to expand into the form:
      *
      * <p>type KlassAlias = Klass; const KlassAlias = typeof Klass;
+     *
+     * <p>With incremental clutz it is possible that the type is not present. In such cases one
+     * needs to pass 'null' and an explicit non-null alternativeAliasName will be used.
      */
-    private void visitTypeValueAlias(String unqualifiedName, ObjectType otype) {
-      String typeName = Constants.INTERNAL_NAMESPACE + "." + otype.getDisplayName();
+    private void visitTypeValueAlias(
+        String unqualifiedName, @Nullable ObjectType otype, @Nullable String alternativeAliasName) {
+      if ((otype == null) == (alternativeAliasName == null)) {
+        throw new IllegalArgumentException(
+            "Visiting symbol "
+                + unqualifiedName
+                + " with aliased type and alternative name or neither.");
```

Comment:
also known as I wish java had union types.

Reply:
I think I'd overload the function maybe.